app/controllers/ruleset_controller.rb

```diff
@@ -3,23 +3,53 @@
 # TODO: Separate this out into a controller per ruleset
 class RulesetController < ApplicationController
   before_action :set_ruleset_type, only: %i[new create]
+  before_action :clear_errors
 
   def new
     render @ruleset_name
   end
 
   def create
+
+    # If no ruleset has been specified (someone has bypassed it somehow)
+    # redirect to the home page
+    if @ruleset_name.nil?
+      redirect_to root_path
+      return
+    end
+
+    # If a file hasn't been uploaded yet, ask for one.
     if ruleset_params[:oas_file].nil?
       flash[:error] = "Please upload a file"
       return render @ruleset_name
     end
 
-    # Here we can call our 42 crunch or spectral service
-    if @ruleset_name == 'government_ruleset'
-      linter_output = Linters::Spectral.new(file: ruleset_params[:oas_file]).lint_to_json
-      render(json: linter_output)
+    # If it is not a valid JSON file, ask again.
+    if ruleset_params[:oas_file].content_type != "application/json"
+      flash[:error] = "This is not a valid JSON file"
+      return render @ruleset_name
+    end
+
+    # Make the API calls in a begin/rescue block so that
```

Comment:
Also a good shout, I think it may be worth down the line splitting out the api depending on how we evolve the app